about/release_policy.rst

```diff
@@ -3,48 +3,107 @@
 Godot release policy
 ====================
 
+Godot's release policy is in constant evolution. What is described below is
+intended to give a general idea of what to expect, but what will actually
+happen depends on the choices of core contributors, and the needs of the
+community at a given time.
+
 Godot versioning
 ----------------
 
-Godot uses ``major.minor.patch`` version numbering. However, it does not
-strictly follow `Semantic Versioning <https://semver.org/>`__. This means that
-releases considered "semver-minor" by that standard (such as 3.1 -> 3.2) will
-most likely introduce breaking changes. Still, there won't be as many breaking
-changes as a "semver-major" version bump such as 3.2 -> 4.0.
+Godot loosely follows `Semantic Versioning <https://semver.org/>`__ with a
+``major.minor.patch`` versioning system, albeit with an interpretation of each
+term adapted to the complexity of a game engine:
+
+- The ``major`` version is incremented when major compatibility breakages happen
+  which imply significant porting work to move projects from one major version
+  to another.
+
+  For example, porting Godot projects from Godot 2.1 to Godot 3.0 required
+  running the project through a conversion tool, and then performing a number
+  of further adjustments manually for what the tool could not do automatically.
+
+- The ``minor`` version is incremented for feature releases which do not break
+  compatibility in a major way. Minor compatibility breakage in very specific
+  areas *may* happen in minor versions, but the vast majority of projects
+  should not be affected or require significant porting work.
+
+  The reason for this is that as a game engine, Godot covers many areas such
+  as rendering, physics, scripting, etc., and fixing bugs or implementing new
+  features in a given area may sometimes require changing the behavior of a
+  feature, or modifying the interface of a given class, even if the rest of
+  the engine API remains backwards compatible.
+
+.. tip::
+
+    Upgrading to a new minor version is therefore recommended for all users,
+    but some testing is necessary to ensure that your project still behaves as
+    expected in a new minor version.
 
-In the interest of stability and usability, patch releases may occasionally
-introduce small breaking changes as well. When repackaging Godot projects (e.g.
-in a Flatpak), make sure to always use the same patch version as the one used to
-initially export the project.
+- The ``patch`` version is incremented for maintenance releases which focus on
+  fixing bugs and security issues, implementing new requirements for platform
+  support, and backporting safe usability enhancements. Patch releases are
+  backwards compatible.
+
+  Patch versions may include minor new features which do not impact the
+  existing API, and thus have no risk of impacting existing projects.
```

Comment:
This is up for debate if we want to follow semver more strictly, but do we really want to prevent ourselves from e.g. adding a new method binding for scripts in stable branches?

Personally I'm of the opinion that this kind of change is safe, and that if it's needed by users *now*, we may as well provide the new API. It does mean that once users start using the new API in a patch release, they can't go back to older releases in that same branch without changing their code back.